config/StoredProcedures/GetAllInventory.sql

```diff
@@ -30,4 +31,4 @@ on s1.ProductID = s2.ProductID;
 
 END $$
 
-DELIMITER ;
\ No newline at end of file
+DELIMITER ;
```

Comment:
This looks like it works, but might contain some unnecessary joins. I believe the following is equivalent and uses less joins:

SELECT  S1.ProductID, S1.ProductName, S1.TotalQuantityAvailable, S1.LastRunDate, S1.LastRun, S2.InitialQuantity AS LastRunInitialQuantity
FROM (Select Pr.ProductID, Pr.Name AS ProductName, SUM(QuantityAvailable) AS TotalQuantityAvailable, MAX(R.DateCreated) AS LastRunDate, MAX(R.RunID) AS LastRun
	  FROM Products Pr LEFT JOIN Piles Pi ON Pr.ProductID = Pi.ProductID LEFT JOIN Runs R ON Pi.PileID = R.PileID
	  GROUP BY Pr.ProductID
	 ) AS S1
LEFT JOIN
	 (Select RunID, InitialQuantity
	  FROM Runs
	 ) AS S2
ON S1.LastRun = S2.RunID;

I played around with this mostly for my own benefit, so use whichever works for you.